### DISCUSSION TITLE
Brave browser not playing nicely?

### DISCUSSION THREAD

**Discussion by jgclark** [ttscoff/bunch-docs#11] [Q&A] [answered]
Running 1.4beta, I got this error when running `@@` (hide all):
<img width="374" alt="Screenshot 2021-02-08 at 21 38 07" src="https://user-images.githubusercontent.com/1671754/107363327-e2295200-6ad1-11eb-9fa9-135c8c76c439.png">

The relevant part of the log says:
```
2021-02-08 21:38:08: 🔴{
    NSAppleScriptErrorAppName = "System Events";
    NSAppleScriptErrorBriefMessage = "AppleEvent timed out.";
    NSAppleScriptErrorMessage = "System Events got an error: AppleEvent timed out.";
    NSAppleScriptErrorNumber = "-1712";
    NSAppleScriptErrorRange = "NSRange: {36, 51}";
}
```

This may well not be Bunch's fault, but I'm reporting it to see if others are seeing this. If so, I'll let Brave know.

**Comment by ttscoff** [comment, marked as answer]
The next update actually removes the AppleScript portion of the Hide All function. It's too slow and unreliable. If an app doesn't respond to the NSWorkspace request to hide, it's just going to be ignored. I'll test this with Brave and see if using the AppleScript wasn't just overkill to begin with.

**Comment by ttscoff** [comment]
Are you still seeing this issue with the latest release?

**Comment by jgclark** [comment]
On Build 69 this does indeed work with Brave. Thanks.
However, Finder windows disappear briefly and then reappear. I now can't
remember what the previous behaviour was for Finder.

**Comment by ttscoff** [comment]
Regarding Finder, is this in a Bunch where they should hide?

**Comment by ttscoff** [comment]
Oh yeah, it's @@.

macOS always wants something visible, so the last thing that hides usually resurfaces. What's most effective is to @Focus a single app.

**Reply by ttscoff** [reply]
What I've been doing if I want the desktop cleared before new apps launch is to launch the first of the new apps and @focus it, then launch the others. Anything that hasn't already launched at the time the "Activate hiding other apps" command is sent won't hide, so it has the same effect as running `@@` before the first app. I'll probably update the documentation as such, because the `hide all` behavior grinds up against standard system behavior and it's not something I could easily work around.

**Comment by jgclark** [comment]
That'll be why I didn't notice it in normal use. When I tried the edge case
of `@@` on its own, that's when I spotted it. All is good.
